src/Adapter/Product/Options/RedirectTargetProvider.php

```diff
@@ -113,7 +112,7 @@ public function getRedirectTarget(
     private function getProductTarget(int $redirectTargetId): ProductRedirectTarget
     {
         $languageId = (int) $this->legacyContext->getLanguage()->id;
-        $product = $this->productPreviewRepository->getPreview(
+        $product = $this->productImageRepository->getProductCoverUrl(
```

Comment:
Here since both name and image are needed you would use the `ProductPreviewRepository` service (that needs to be re-introduced)